deploy/scripts/src/cri-kiwi/BAV.ts

```diff
@@ -0,0 +1,174 @@
+
+  res = group('B01_BAV_03_Continue POST', () =>
+    timeRequest(() =>
+      res.submitForm({
+        params: { tags: { name: 'B01_BAV_03_Continue' } },
+        submitSelector: '#landingPageContinue'
+      }),
+    { isStatusCode200, ...pageContentCheck('Enter your account details') }))
+
+  sleepBetween(1, 3)
+
+  res = group('B01_BAV_04_BankDetails POST', () =>
+    timeRequest(() =>
+      res.submitForm({
+        fields: {
+          sortCode: testSortCode,
+          accountNumber: testAccountNumber
+        },
+        params: { tags: { name: 'B01_BAV_04_BankDetails' } },
+        submitSelector: '#continue'
+      }),
+    { isStatusCode200, ...pageContentCheck('Check your details match with your bank or building society account') }))
+
+  sleepBetween(1, 3)
+
+  res = group('B01_BAV_05_CheckDetails POST', () =>
+    timeRequest(() =>
+      res.submitForm({
+        params: { tags: { name: 'B01_BAV_05_CheckDetails' } },
+        submitSelector: '#submitDetails'
+      }),
+    {
+      'verify url body': (r) =>
+        (r.url).includes(clientId)
+    }))
+  const codeUrl = getCodeFromUrl(res.url)
+
+  sleepBetween(1, 3)
+
+  res = group('B01_BAV_06_SendAuthorizationCode POST', () =>
+    timeRequest(() => http.post(env.BAV.target + '/token', {
+      grant_type: 'authorization_code',
+      code: codeUrl,
+      redirect_uri: env.BAV.ipvStub + '/redirect?id=bav'
+    }, {
+      tags: { name: 'B01_BAV_06_SendAuthorizationCode' }
+    }),
+    { isStatusCode200, 'verify response body': (r) => (r.body as string).includes('access_token') }))
+
+  const accessToken = getAccessToken(res)
+
+  sleepBetween(1, 3)
+
+  const authHeader = `Bearer ${accessToken}`
+  const options = {
+    headers: { Authorization: authHeader },
+    tags: { name: 'B01_BAV_07_SendBearerToken' }
+  }
+  res = group('B01_BAV_07_SendBearerToken POST', () =>
+    timeRequest(() => http.post(env.BAV.target + '/userinfo', {}, options),
+      { isStatusCode200, 'verify response body': (r) => (r.body as string).includes('credentialJWT') }))
+  iterationsCompleted.add(1)
+}
+
+function getClientID (r: Response): string {
+  const clientId = r.json('clientId')
+  if (clientId !== null && typeof clientId === 'string') return clientId
+  fail('Client ID not found')
+}
+
+function getCodeFromUrl (url: string): string {
+  const code = url.match(/code=([^&]*)/)
+  if (code?.[1] != null) return code[1]
+  fail('Code not found')
+}
+
+function getAccessToken (r: Response): string {
+  const accessToken = r.json('access_token')
+  if (accessToken !== null && typeof accessToken === 'string') return accessToken
+  fail('AccessToken not found')
+}
+
+function getAuthorizeauthorizeLocation (r: Response): string {
+  const authorizeLocation = r.json('AuthorizeLocation')
+  if (authorizeLocation !== null && typeof authorizeLocation === 'string') return authorizeLocation
+  fail('AuthorizeLocation not found')
+}
```

Comment:
These are the same functions as used in `cri-kiwi/f2f-cic.ts`. Can these be split out into a utility file?